src/hotspot/share/code/stubs.cpp

```diff
@@ -238,7 +238,7 @@ void StubQueue::verify() {
 
 
 void StubQueue::print() {
-  MutexLocker lock(_mutex, Mutex::_no_safepoint_check_flag);
+  ConditionalMutexLocker lock(_mutex, _mutex != nullptr, Mutex::_no_safepoint_check_flag);
```

Comment:
Why is this one now a `ConditionalMutexLocker`?

Reply:
I think there was a test failure that indicated we enter here from some error path? Let me reproduce it again.

Reply:
Got one: `vmTestbase/nsk/jvmti/AddCapabilities/addcaps003/TestDescription.java`.

The reproducer is as simple as:

```
% build/macosx-aarch64-server-fastdebug/images/jdk/bin/java -XX:+PrintInterpreter
...
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  Internal Error (/Users/shipilev/Work/shipilev-jdk/src/hotspot/share/runtime/mutexLocker.hpp:237), pid=85876, tid=8451
#  assert(mutex != nullptr) failed: null mutex not allowed
#
```

Reply:
Thanks for chasing down the reason.